### PR TITLE
Update E2E test suite creator and runner

### PR DESCRIPTION
**Pull request by SEAJang**
# Pull Request Key Information

## Reason for change

End to end tests were outdated with the change from V3 to V4 

## What was changed

Updated the end-to-end test suit creator, and the running script and remade 4 basic E2E tests

## Intended Purpose

Update the end-to-end tests from v3 to v4. 

## Testing Completed

Manually tested to see if the scripts ran, and verified the E2E results. 
[V4-simple_all_repairable_3c176cb78d0106ec77c2951c7e1bae09b9dcd831.log](https://github.com/user-attachments/files/15907518/V4-simple_all_repairable_3c176cb78d0106ec77c2951c7e1bae09b9dcd831.log)

All existing unit tests run
[results.txt](https://github.com/user-attachments/files/15907517/results.txt)

## Level of version change required

Patch

## Additional Information

More E2E tests will be added in future patches.

### CODE REVIEW

**Review by ThomasGalesloot — APPROVED** [review]
Great work so far - I think this has a little more updating needed to be great

**Review by ThomasGalesloot — CHANGES_REQUESTED** [review]
Meant to Request changes not approve sorry

File: LDAR_Sim/testing/end_to_end_testing/e2e_suite_runner.py [hunk 1]
```diff
@@ -0,0 +1,236 @@
...
+e2e_test_dir: Path = Path(os.path.dirname(os.path.realpath(__file__)))
+root_dir: Path = e2e_test_dir.parent.parent
+src_dir: Path = root_dir / "src"
+tests_dir: Path = e2e_test_dir / "test_suite"
+os.chdir(root_dir)
+# Add the source directory to the import file path to import all LDAR-Sim modules
+sys.path.insert(1, str(src_dir))
+
+if __name__ == "__main__":
+    from initialization.initialize_infrastructure import initialize_infrastructure
```
[the remark this answers is not in the document]
Comment (SEAJang) [reply LDAR_Sim/testing/end_to_end_testing/e2e_suite_runner.py:41]: The best thing that could be done was to alphabetize them.

File: LDAR_Sim/testing/end_to_end_testing/testing_utils/result_verification.py [hunk 2]
```diff
@@ -0,0 +1,203 @@
...
+        )
+    except subprocess.CalledProcessError:
+        # handle error if Git command fails
+        git_commit_hash = None
+    return git_commit_hash
+
+
+def compare_params(out_dir, expected_dir) -> Literal["Success", "Failure"]:
+    # Read in parameters and expected parameters
+    with open(out_dir / "parameters.yaml", "r") as params_file:
```
[the remark this answers is not in the document]
Comment (SEAJang) [reply LDAR_Sim/testing/end_to_end_testing/testing_utils/result_verification.py:52]: This does exist, however, we've had the issue where this file is usually one of the key files that is changed during an update. And the point of this E2E is to get the same results even if more parameters are added. 
This code is commented out in the main testing script, but if it's better removed, please let me know.

**Comment by SEAJang** [comment]
New E2E Test results files:
[V4-simple_all_repairable_9ed4958239661f54f2456b4d0a532e49706d9d6b.log](https://github.com/user-attachments/files/15961995/V4-simple_all_repairable_9ed4958239661f54f2456b4d0a532e49706d9d6b.log)
[V4-simple_all_non_repairable_9ed4958239661f54f2456b4d0a532e49706d9d6b.log](https://github.com/user-attachments/files/15961996/V4-simple_all_non_repairable_9ed4958239661f54f2456b4d0a532e49706d9d6b.log)

All unit existing unit tests pass:
[results.txt](https://github.com/user-attachments/files/15962001/results.txt)

**Comment by SEAJang** [comment]
All existing unit tests pass. 
[results.txt](https://github.com/user-attachments/files/15983026/results.txt)

All E2E tests pass - ran on a different machine. 
[V4-simple_all_non_repairable_9c4b0ae71517cb6beb0f6e2d73f09e15faa176cf.log](https://github.com/user-attachments/files/15983235/V4-simple_all_non_repairable_9c4b0ae71517cb6beb0f6e2d73f09e15faa176cf.log)
[V4-simple_all_repairable_9c4b0ae71517cb6beb0f6e2d73f09e15faa176cf.log](https://github.com/user-attachments/files/15983236/V4-simple_all_repairable_9c4b0ae71517cb6beb0f6e2d73f09e15faa176cf.log)
[V4-simple_mixed_emissions_9c4b0ae71517cb6beb0f6e2d73f09e15faa176cf.log](https://github.com/user-attachments/files/15983237/V4-simple_mixed_emissions_9c4b0ae71517cb6beb0f6e2d73f09e15faa176cf.log)

**Review by ThomasGalesloot — APPROVED** [review]
Great work - Just address the few small comments and then good to merge. AS a sidenote - we need to think if we really want to compare every output file or just the summary ones.

File: LDAR_Sim/testing/end_to_end_testing/test_suite/V4-simple_all_non_repairable/params/M_OGI.yaml [hunk 3]
```diff
@@ -0,0 +1,15 @@
+parameter_level: methods
+version: "4.0"
+method_name: OGI
+deployment_type: mobile
+measurement_scale: component
+sensor:
+  type: OGI_camera_zim
```
[the remark this answers is not in the document]
Comment (SEAJang) [reply LDAR_Sim/testing/end_to_end_testing/test_suite/V4-simple_all_non_repairable/params/M_OGI.yaml:7]: The OGI sensor if not provided,comes with a default now.

File: LDAR_Sim/testing/end_to_end_testing/e2e_test_case_creator.py [hunk 4]
```diff
@@ -0,0 +1,277 @@
...
+            gc.collect()
+            print(rm.FIN_SIM_SET.format(simulation_number=simulation_number))
+
+        # -- Batch Report --
+        print(rm.BATCH_CLEAN.format(batch_count=0))
+        summary_stats_manager.gen_summary_outputs(False)
+        non_baseline_progs = get_non_baseline_prog_names(programs, base_program)
+        summary_stats_manager.gen_cost_summary_outputs(non_baseline_progs)
+    os.chdir(root_dir)
+    shutil.move(params_dir, test_case_dir)
```
[the remark this answers is not in the document]
Comment (SEAJang) [reply LDAR_Sim/testing/end_to_end_testing/e2e_test_case_creator.py:275]: You do keep the original test case parameters.

**Comment by SEAJang** [comment]
[results.txt](https://github.com/user-attachments/files/16198693/results.txt)

**Comment by SEAJang** [comment]
[results.txt](https://github.com/user-attachments/files/16198775/results.txt)

**Pull request merged**